jung-algorithms/src/main/java/edu/uci/ics/jung/layout/spatial/Node.java

```diff
@@ -14,7 +14,15 @@
 public class Node<T> {
 
   private static final Logger log = LoggerFactory.getLogger(Node.class);
-  public static final double THETA = 0.5f;
+
+  /**
+   * threshold value for determining whether to use the forces in an inner node as a summary value,
+   * or to descend the quad tree to another inner node or leaf node. theta should be between 0 and
+   * 1, with 0.5 as a commonly selected value. Note that a theta value of 0 will result in no
+   * optimization in the BarnesHutQuadTree and all force visitors will descend to a leaf node
+   * instead of using an inner node summary force vector.
+   */
+  public static final double DEFAULT_THETA = 0.5;
```

Comment:
The builders are new (added so I could set the theta someplace). They have no javadocs at all yet. I made theta settable for a unit test (explained below). I am not sure we want users to ever touch that value. Perhaps testing over various size graphs will expose a need though.
WIth the current visibility, it is public so that the layout algorithms can see it (and be the place where it is set). The approach will probably change when the LayoutAlgorithms get the builders you want to write.
I should mention that the settable theta now allows the unit test to directly compare the results of the BH version with the non-BH version. The unit tests include that now. The 'Iterator BH' versions really can go away, although they are still present and may serve a purpose as test code. I fear that I'm guilty of hubris when I delete test code.

Reply:
It may be worth noting that I do not advocate names like SpringFRVisitorLayoutAlgorithm should remain as such. At some point, we will probably just make the BHTree usage something that is available (maybe also a default) in the builder for those layout algorithms.
One challenge in making builders for the layout algorithms is that there is a lot of variety and few shared features. The Iterative ones share some parameters, and the BHTree-able ones do too.
Perhaps (instead of the layoutalgorithm knowing about theta) the builder for the BHTree part of a layout algorithm should take in some object with BHtree settings (like theta). I think that could make it more clear (instead of 'what's this theta thing, some angle?').

FWIW I have looked at KKLayoutAlgorithm a couple of times to see how hard it would be to break up the polynomial that calculates energy so that it could use force estimations from a BHTree.

Reply:
I'd like to model the builders that I wrote (in layout model and in the BHTree code) after the design that you choose for the layout algorithms (and wherever else you create builders in JUNG). I think we'd agree that JUNG should offer users a consistent style for builders. I consider these builders as sort of 'place holders' for now, until I see the best design for them, and the style of their javadocs.